### PR TITLE
Phelan dev

### PR DESCRIPTION
**Pull request by phelanmccall**

### CODE REVIEW

**Comment by allen138** [comment]
will resubmit when package-lock is ignored

**Comment by phelanmccall** [comment]
Resolved conflicts.

**Review by allen138 — APPROVED** [review]
lgtm

File: client/src/components/Form.js [hunk 1]
```diff
@@ -41,12 +41,13 @@ class Form extends Component {
 
     const data = { ...this.state.data };
     data[input.name] = input.value;
+    console.log(data)
```
Comment (jmckenney) [inline client/src/components/Form.js:44]: Don't forget to remove. Likely there is an eslint rule complaining about this as well:)

File: client/src/components/Form/form.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
+import React, { Component } from "react";
+import Input from "../LoginForm/loginInput.js/input";
+import Joi from "joi-browser";
```
Comment (jmckenney) [inline client/src/components/Form/form.js:3]: Worth considering, not a blocker at all: https://bundlephobia.com/result?p=joi-browser@13.4.0

File: client/src/components/RegisterForm.js [hunk 3]
```diff
@@ -26,7 +26,10 @@ class RegisterForm extends Form {
 
   doSubmit = async () => {
     try {
-      await userService.register(this.state.data);
+      console.log(this.state.data)
```
Comment (jmckenney) [inline client/src/components/RegisterForm.js:29]: Another place to consider cleaning out in an upcoming PR.

File: client/src/components/RegisterForm.js [hunk 4]
```diff
@@ -26,7 +26,10 @@ class RegisterForm extends Form {
...
+      await userService.register(this.state.data).then((user) => {
```
Comment (jmckenney) [inline client/src/components/RegisterForm.js:30]: Excellent use of await here.

File: client/src/components/RegisterForm/registerForm.js [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+class RegisterForm extends Form {
+  state = {
+    data: { firstname: "", lastname: "", email: "", password: "" },
+    errors: {}
+  };
+  schema = {
+    firstname: Joi.string()
+      .required()
+      .label("Firstname"),
```
Comment (jmckenney) [inline client/src/components/RegisterForm/registerForm.js:19]: Label is the visible text or the name of the field? If visible should likely be `First Name`

**Pull request merged**